instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/tracer/HttpClientTracer.java

```diff
@@ -183,8 +183,14 @@ private void setUrl(AttributeSetter setter, REQUEST request) {
       URI url = url(request);
       if (url != null) {
         netPeerAttributes.setNetPeer(setter, url.getHost(), null, url.getPort());
-        setter.setAttribute(SemanticAttributes.HTTP_URL, url.toString());
+        if (url.getUserInfo() != null) {
+          setter.setAttribute(
+              SemanticAttributes.HTTP_URL, url.toString().replace(url.getUserInfo() + '@', ""));
```

Comment:
Instead of using string replacement (which can replace more than once) consider creating a new `URI` without the user info part:

```java
new URI(
    url.getScheme(),
    null,
    url.getHost(),
    url.getPort(),
    url.getPath(),
    url.getQuery(),
    url.getFragment());
```

Reply:
Thanks for your suggestion! I'll fix them all.